### ISSUE TITLE
Special Building interaction

### ISSUE THREAD

**Issue by agrignard** [CityScope/CS_Cooper-Hewitt#86]
The fact to have special building triggering some vizu  was initially  a quick hack to test the connectivity of the scanning and the simulation but I think it's pretty cool from a user point of view

Right now we have 

Building 20 (pyramid): trigger color 
Building 21 (empire): trigger network
Building 22 (flat one): trigger agent

It can be nice to have some other special one, 
- collision potential 
- show different background

And if you have other idea we can think about other trigerring

@guadalupebabio what kind of special building do we have?

**Comment by LAAP** [comment]
@agrignard , I think that we should remove this one:

Building 22 (flat one): trigger agent

And the "effects" should only last few seconds

**Comment by agrignard** [comment]
Flat one (22) is now triggering the collision potential

**Comment by agrignard** [comment]
I close this one for now. The temporal appearance needs to be discusssed and is not that easy to implement.

**Comment by agrignard** [comment]
I reopen this issue as a bug/enhancement as @LAAP has requested that the special effect only last for a certain period. I am in a why not mode, I am not against keeping the effect as long as the building is here but it also makes sense to make it disappears automatically, so let's have the 2 possibility and we'll discuss at the end which one we keep. That's why it's a bug/enhancement (we need to have this option before the deployment)

@yasushisakai: To have a temporary special effect let's customize the GridInteractionAnimation class to add some special effect for some specific building.

The idea would be to keep the gridanimation as it is but had the following feature for the empire state building (and potentially for some other)

As soon as the empire state building appears, we show the network for a specific timing (bigger than DURATION) maybe 10*DURATION or what so ever.

I think it's a good design to use the structure you used for the gridAnimation , it help us to customized some special effect and maybe add more easily later with all the special building

**Comment by agrignard** [comment]
Ok so I just didi it in https://github.com/CityScope/CS_Cooper-Hewitt/commit/3ee0cabb4e356ef25e126231c0b89627b771c1ac

@yasushisakai I did some slight cahnge in take and put in orde to know which block has been put or taken, then there is a special activate that take more or less the same concet of the activate but with a customization to activate a special boolean.

With this modify **ALL** the special effect (color, network, collision) are now during 10 sec max (if you remove the piece you don’t see the effect anymore)

This can be discussed on how long we want to keep it, if we want to keep etc. For now this is the behavior in the master

@yasushisakai please review this change to be sure that it doesn't change your animation features and please stress test it on the table. I test it with the virtualtable I think it 's fine but we need to be sure on the physical table.

**Comment by agrignard** [comment]
Even if I said it shouln't be merge it has been merge so right now after this commit
https://github.com/CityScope/CS_Cooper-Hewitt/commit/3b023ba7c5a621cbea33664fc00a6e37d1748e2f

the special effect that was working this morning is rebroken. (I think the diff has not been made exactly) I will try to have a look but we have to be more careful

PLEASE be careful when merging a version especially at this time of the project.

**Comment by agrignard** [comment]
One first thing that was not taken in account in the merge (the change I made this morning were not taken in account in this merge)
https://github.com/CityScope/CS_Cooper-Hewitt/commit/7c854d9358715e35097493f042fe8d06fe0eea61


Now th eproblem is when i remove a block the effect is still here it was not the case in this commit
https://github.com/CityScope/CS_Cooper-Hewitt/commit/3ee0cabb4e356ef25e126231c0b89627b771c1ac

**Comment by agrignard** [comment]
updateGridFromUDP has been modify and it not doing the same thing as it was in 
3ee0cab

**Comment by yasushisakai** [comment]
The table is in now full restoration and I don't have my computer to test.
I will revisit this when I have either.

**Comment by agrignard** [comment]
Temporally broken (was working in 3ee0cab)

To Do: merge PR92 and 3ee0cab however #84 is more prioritary than #86

Once #84 is closed it shouldn't be difficult to reimplement #86 maybe by separating the update of the building on grid and the gridAnimation

**Comment by yasushisakai** [comment]
We have had a follow-up meeting w/ @LAAP and Kent.

The special buildings interaction was decided to be removed.

The reasoning is "it's confusing."

This issue is to have interaction,
I will have a separate isssue to remove it.

**Comment by agrignard** [comment]
ok so in this case let's not focus on this one and keep it as a potential enhancement for a future project. 

Anyway this was just initially a quick hack to show that the scanning was working. it would have been nice to know it before. 

I will remove what I did this morning so we can focus on issue #84

**Comment by yasushisakai** [comment]
Yes, at least we have the attempt in the history and this issue. It is not lost.

**Comment by agrignard** [comment]
Let's close this issue (see issue #94)

**Comment by LAAP** [comment]
Hi @agrignard @yasushisakai @aberke 

The last 5% of the special building interaction is to have:

1) Interaction with the buildings. Now, special buildings don't have a clear population or interaction.

2) Kent's requirement: the "vomiting" of agents as soon as you move a building. 

What I suggest to solve the 2 issues:

- To have a new species of agents (anti-zombie) that goes from the new buildings to the special buildings every time that you place the building (We can limit the number of agents on each push in order to have a stable simulation)

- Everytime that we place a building we can push a % of the population on the building (maybe 1/3).

- We can keep the flow, or the anti-zombies can die in the special buildings

-If the special buildings are in zombie land, the anti-zombie will go to zombie land.

**Comment by guadalupebabio** [comment]
We have never talked about pieces 22 and 23 (parks), is there anything for them in processing?. Maybe the wave effect that was tested today in unity can be applied to this two pieces. Anyways for the wave effect we have different options:

a. Pyramid & Tower
B. Parks
C. Both
D. None of the above (delete)

**Comment by LAAP** [comment]
Hi @guadalupebabio ,

I think it should be C;

Not sure about last unity visualization, but we use to have a cool visualization of how the parks transform into parkings and dead fish? (not sure about the fish), @gabrielabila and @maitaneiruretagoyena may know better than me. I think that those visualizations where very cool.

Regarding the "effect", I think that we can use the unity one, as a "Wow!-special-effect", at the special buildings, if we refine it a bit, and if @yasushisakai thinks that it matches with the "overall symphony".

In parallel, I think that we should have the "agent vomiting effect", in all the buildings, so we can have a flow between the new placed buildings and the "special buildings", isn't @agrignard ?   ;)

Cheers!

**Comment by agrignard** [comment]
> We have had a follow-up meeting w/ @LAAP and Kent.
> 
> The special buildings interaction was decided to be removed.
> 
> The reasoning is "it's confusing."
> 

This is concerning the special building interaction, it has been decided to remove it from the simulation so some choice has been made according to that. That means that right now the interaction doesn't take in account the spatial building.

Some architectural choice has been made since so it's not that easy to realive this feature and especially dangerous to change so close to the deployment but let's see...

**Comment by LAAP** [comment]
Just for the record: Kend didn't say "let's kill the special buildings". What Kent said is that the output of the interaction was confusing = triggering permanently the grid, the color, the dots, and the agents, was cool, however, was confusing.
The message was that "it is OK to have special buildings", however, "what you trigger with them needs to be consistent with the overall symphony". So this applies to all the interactions in the table. That is the reason why we where doing today all the design work with the processing output and it also applies to the Unity option that @guadalupeabio is suggesting. Does it makes sense?

**Comment by agrignard** [comment]
0 issue the d-day!
